pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -339,3 +339,10 @@ def test_setitem_boolean_mask(self, mask_type, float_frame):
         expected = df.copy()
         expected.values[np.array(mask)] = np.nan
         tm.assert_frame_equal(result, expected)
+
+    def test_setitem_only_false_indexer_dtype_changed(self):
+        # GH: 37550
+        df = DataFrame({"a": ["a"], "b": [1], "c": [1]})
+        df.loc[[False], ["b"]] = 10 - df["c"]
+        expected = DataFrame({"a": ["a"], "b": [1], "c": [1]})
```

Comment:
can you *also* test the rhs as just a scalar integer (e.g. 10 - 1), apparently this does the same behavior.

Reply:
Done

Reply:
Runs a different code path, that is the reason we did not have problems with a scalar :)

Reply:
sure that's fine, but just locking it down with a test